tests/test-proxy.rs

```diff
@@ -52,33 +52,29 @@ fn hello_request_method(req: Request, mut res: Response) {
 }
 
 #[test]
-fn get_on_http_server() {
-    let _ = env_logger::init();
-
-    with_server(hello_request_method, 8081, &|port| {
-        let client = Client::new();
-        let url = url(port);
-        let mut res = client.get(&url).send().unwrap();
-        assert_eq!(res.status, hyper::Ok);
-
-        let mut body = String::new();
-        res.read_to_string(&mut body).unwrap();
-        assert_eq!(body, "hello GET");
-    });
-}
+fn method_on_http_server() {
+    use hyper::method::Method;
+    use std::str::FromStr;
 
-#[test]
-fn delete_on_http_server() {
     let _ = env_logger::init();
 
-    with_server(hello_request_method, 8082, &|port| {
-        let client = Client::new();
-        let url = url(port);
-        let mut res = client.delete(&url).send().unwrap();
-        assert_eq!(res.status, hyper::Ok);
+    let methods = vec!("GET", "DELETE", "PATCH", "OPTIONS", "POST", "PUT", "TRACE", "HEAD", "CONNECT", "GARBAGE");
```

Comment:
By `GARBAGE` I meant a non-standard request.

I agree with you. I'd like to work on requests with body after that.

> we will reject non-standard http requests because there is no way to know how to handle them.

What do you mean with non-standard http requests? The test with `GARBAGE` is non-standard but works nevertheless.